guides/index-template.md

```diff
@@ -0,0 +1,304 @@
+            ""pages"": { ""type"": ""integer"" }
+        }
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponse}") // Create response: {"acknowledged":true}
+```
+
+Now, when you create an index that matches the `books-*` pattern, OpenSearch will automatically apply the template's settings and mappings to the index. Let's create an index named books-nonfiction and verify that its settings and mappings match those of the template:
+
+```csharp
+var createResponse = client.Indices.Create<StringResponse>("books-nonfiction", PostData.String(@"{
+    ""settings"": {
+        ""number_of_shards"": 3,
+        ""number_of_replicas"": 0
+    },
+    ""mappings"": {
+        ""properties"": {
+            ""title"": { ""type"": ""text"" },
+            ""author"": { ""type"": ""text"" },
+            ""published_on"": { ""type"": ""date"" },
+            ""pages"": { ""type"": ""integer"" }
+        }
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponse}"); // Create response: {"acknowledged":true,"shards_acknowledged":true,"index":"books-nonfiction"}
+```
+
+
+## Multiple Index Templates
+If multiple index templates match the index's name, OpenSearch will apply the template with the highest `priority`. The following example creates two index templates named `books-*` and `books-fiction-*` with different settings:
+
+```csharp
+var createResponseOne = client.Indices.PutTemplateV2ForAll<StringResponse>("books", PostData.String(@"{
+    ""index_patterns"": [""books-*""],
+    // ""priority"": 1,
+    ""settings"": {
+        ""number_of_shards"": 3,
+        ""number_of_replicas"": 0
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponseOne}"); // Create response: {"acknowledged":true}
+
+var createResponseTwo = client.Indices.PutTemplateV2ForAll<StringResponse>("books", PostData.String(@"{
+    ""index_patterns"": [""books-fiction-*""],
+    ""priority"": 2,
+    ""settings"": {
+        ""number_of_shards"": 3,
+        ""number_of_replicas"": 0
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponseTwo}"); // Create response: {"acknowledged":true}
+```
+
+When we create an index named `books-fiction-romance`, OpenSearch will apply the `books-fiction-*` template's settings to the index:
+
+```csharp  
+var createResponse = client.Indices.Create<StringResponse>("books-fiction-romance", PostData.String(@"{
+    ""settings"": {
+        ""number_of_shards"": 3,
+        ""number_of_replicas"": 0
+    },
+    ""mappings"": {
+        ""properties"": {
+            ""title"": { ""type"": ""text"" },
+            ""author"": { ""type"": ""text"" },
+            ""published_on"": { ""type"": ""date"" },
+            ""pages"": { ""type"": ""integer"" }
+        }
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponse}") // Create response: {"acknowledged":true,"shards_acknowledged":true,"index":"books-fiction-romance"}
```

Comment:
Creating the index explicitly specifying the mappings and settings defeats the point of the template, the more relevant code would be:
```suggestion
client.Indices.Create("books-fiction-romance");
var getResponse = client.Indices.Get("books-fiction-romance");
Console.WriteLine(getResponse.Indices["books-fiction-romance"].Settings.NumberOfShards); // 1
```